packages/nimble-components/src/tabs/specs/scrollable-tabs.md

```diff
@@ -0,0 +1,110 @@
+# Scrollable tabs
+
+## Problem Statement
+
+There are some use-cases that call for many tabs, and in cases where the width of all of the tabs exceed the screen space, they will bleed off screen (and potentially make the page scrollable to see the other tabs). To prevent this, we will introduce "scrollable" tabs, which will enforce the tabs to be constrained to the given space of the `Tabs` component and offering buttons to allow a user to navigate through the whole set of tabs.
+
+### Out of scope of this HLD
+
+Any decision/API around limiting the width of a particular tab. This is ultimately an orthogonal concern to the UX this HLD is presenting.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [Issue 1509: Scrollable Tabs](https://github.com/ni/nimble/issues/1509)
+-   [Figma design](https://www.figma.com/design/PO9mFOu5BCl8aJvFchEeuN/Nimble_Components?node-id=11133-57220&node-type=instance&t=DlnrdRo7KWYlheFz-0)
+
+## Implementation / Design
+
+![Scrollable Tabs](./spec-images/scrollable-tabs.png)
+
+The following decisions were agreed to in the comments section in the linked issue above:
+
+-   No scrollbar will be shown
+-   Keyboard navigation will work as it does today, but when a tab that isn't fully in view gets focus it should be scrolled into view
+-   When a scroll button is pressed we will scroll either a full "page width" or the remainder of the available space, and tab focus is not changed.
+
+Additionally, the mouse wheel should allow scrolling through the tabs, but instead of scrolling a page at a time, a single scroll "tick" will scroll by one tab.
```

Comment:
It looks like [shoelace doesn't scroll by an entire tab, it scrolls by a single pixel](https://shoelace.style/components/tab-group#scrolling-tabs). Are we sure we want to differ here? Scrolling can generate a lot of "ticks" so it seems like scrolling by entire tabs would make it hard to get fine-grained control over which tabs are visible.

Reply:
I agree...will change.

Reply:
So, I will say that I couldn't get the shoelace example to scroll via the mouse wheel (in Chrome), but I will say that a "single pixel" is _not_ what we want. Playing around with it, 20 pixels at a time feels pretty nice, but there's likely an aesthetic "sweet spot" we will need to have consensus on.

Reply:
I've updated the text to be more vague in terms of precise amounts that will be scrolled, but instead describe the goal of the experience.

Reply:
I added a video below showing what Shoelace looks like on macOS when I scroll with the trackpad and it does seem to go pixel-by-pixel when I move slowly. Maybe a trackpad sends more fine-grained events than a mouse scroll wheel? Regardless, I'm fine to keep it vague in the HLD.

For both this question and the one about mobile support, I'm surprised we need to think about the low level events much though. It looks like in the Shoelace shadow DOM there is just a really wide div that contains the tabs, contained in a div that restricts the width to the width of the tab container. The browser handles scrolling through its normal `overflow` and `scrollLeft` attributes. It's the only thing that worries about scrolling the correct amount depending on the platform and input device. Can we do the same?

https://github.com/user-attachments/assets/0c7ad99c-f424-42e5-a587-b35011c700b1

Reply:
I think maybe just using the `deltaY` property of the `WheelEvent` instance will accomplish this (here's an [SO post(https://stackoverflow.com/questions/20110224/what-is-the-height-of-a-line-in-a-wheel-event-deltamode-dom-delta-line)] that dives a bit into the details of each browser and the mode they use for the wheel event).

My suspicion is that we'll see the `deltaY` property have different values dependent on the device being used (and possibly the user agent). For Chrome on Windows with a mouse, for instance, I see a `deltaY` of 100 pixels per scroll tick, which is a bit jumpy IMO, but that's what we would get without intervening.

Reply:
Reopening this based on our conversation about using `overflow: scroll` instead of handling scroll events.

Reply:
Updated.